internal/controller/ols_app_server_assets.go

```diff
@@ -50,19 +59,32 @@ func (r *OLSConfigReconciler) generateOLSConfigMap(cr *olsv1alpha1.OLSConfig) (*
 		}
 		providerConfigs = append(providerConfigs, providerConfig)
 	}
-	OLSRedisMaxMemory := intstr.FromString(OLSAppRedisMaxMemory)
-	OLSRedisMaxMemoryPolicy := OLSAppRedisMaxMemoryPolicy
-	if cr.Spec.OLSConfig.ConversationCache.Redis != (olsv1alpha1.RedisSpec{}) {
+	OLSRedisMaxMemory := intstr.FromString(RedisMaxMemory)
+	OLSRedisMaxMemoryPolicy := RedisMaxMemoryPolicy
+	secretName := RedisSecretName
+	redisConfig := cr.Spec.OLSConfig.ConversationCache.Redis
+	if redisConfig != (olsv1alpha1.RedisSpec{}) {
+		if (redisConfig.CredentialsSecretRef != corev1.LocalObjectReference{}) && (redisConfig.CredentialsSecretRef.Name != "") {
+			secretName = redisConfig.CredentialsSecretRef.Name
+		}
+	}
+	cr.Spec.OLSConfig.ConversationCache.Redis.CredentialsSecretRef.Name = secretName
```

Comment:
See [it](https://github.com/openshift/lightspeed-operator/pull/25/files#r1522071561) here . But , feels like we are updating CR in generateConfigMap() here . We need a clean way to have default value in CRD and use that name.

Reply:
Unfortunately we had redis secret as the [type](https://github.com/openshift/lightspeed-operator/pull/25/files#diff-77e6d7fe1846c7ad02af9efceb28f7be71b1335993f6d36664418fb4c3bec33bR111) `corev1.LocalObjectReference` (i.e to be consistent with rest of the secret references in CRD) for which we cannot set a default value for its nested field `Name` in the CRD itself. So had to do it this way. Since we prefer to have clean and simpler way let me change it `string`. Open to other thoughts too.